server/src/main/java/com/cloud/network/NetworkServiceImpl.java

```diff
@@ -4264,23 +4265,37 @@ public boolean deletePhysicalNetwork(final Long physicalNetworkId) {
         return Transaction.execute(new TransactionCallback<Boolean>() {
             @Override
             public Boolean doInTransaction(TransactionStatus status) {
-                // delete vlans for this zone
-                List<VlanVO> vlans = _vlanDao.listVlansByPhysicalNetworkId(physicalNetworkId);
-                for (VlanVO vlan : vlans) {
-                    _vlanDao.remove(vlan.getId());
-                }
-
-                // Delete networks
-                List<NetworkVO> networks = _networksDao.listByPhysicalNetwork(physicalNetworkId);
-                if (networks != null && !networks.isEmpty()) {
-                    for (NetworkVO network : networks) {
-                        _networksDao.remove(network.getId());
-                    }
-                }
+                disablePhysicalNetwork(physicalNetworkId, pNetwork);
+                deleteVlans();
+                deleteNetworks();
+                deleteIpAddresses();
 
                 // delete vnets
                 _dcDao.deleteVnet(physicalNetworkId);
 
+                boolean x = deleteProviders();
+                if (!x) return false;
```

Comment:
```suggestion
                if (!deleteProviders()) {
                    return false;
                }
```
this is actually according to our coding stadards :(